source/plugins/go/input/lib/applicationinsights.go

```diff
@@ -58,16 +58,21 @@ func init() {
 		logPath = "/var/opt/microsoft/docker-cimprov/log/appinsights_error.log"
 	}
 
+	isTestEnv := os.Getenv("ISTEST") == "true"
+	if isTestEnv {
+		logPath = "./appinsights_error.log"
```

Comment:
all the library files have a logPath inside the container. when we run the unit tests as part of the PR these file paths dont exist since we are not inside a container. therefore we need a different path when running tests